packages/react-widgets/src/widgets/CategoryWidget.js

```diff
@@ -74,6 +75,7 @@ function CategoryWidget(props) {
         joinOperation,
         operation,
         filters,
+        filtersLogicalOperator: source?.filtersLogicalOperator,
```

Comment:
Instead of doing it separately here, what about modifying the `useSourceFilters` hook to return also the operator?
IMO it makes sense to group that as both `filters` and `filtersLogicalOperator` are now required for the filters to be correctly applied.
It would also spare [this selector](https://github.com/CartoDB/carto-react/pull/363/files#diff-b6744b2ddd16d9987012b38f43d5800cd351e453fd471af46f6dbb649801e738R63) as it's already done in the hook.

(And the same comment applies for other widgets of course :slightly_smiling_face:)

Reply:
Ok, we can generalize it in there. Truth is I was searching for a place to get both props from but did not find any, so I'm ok with `useSourceFilters`. 

> It would also spare [this selector](https://github.com/CartoDB/carto-react/pull/363/files#diff-b6744b2ddd16d9987012b38f43d5800cd351e453fd471af46f6dbb649801e738R63) as it's already done in the hook.

I cannot see that link properly, (maybe my version of github is broken) don't know which file is it pointing to

Reply:
> I cannot see that link properly, (maybe my version of github is broken) don't know which file is it pointing to

Sorry for the broken link, I was just pointing at some lines above when you retrieve the source by id:
```js
const source = useSelector((state) => selectSourceById(state, dataSource));
```

Reply:
Refactor done for `useSourceFilters`